gen/definitions/network_groups.yaml

```diff
@@ -0,0 +1,74 @@
+---
+name: Network Groups
+rest_endpoint: /api/fmc_config/v1/domain/{DOMAIN_UUID}/object/networkgroups
+res_description: >-
+  This plural resource manages a bulk of Network Groups.
+  The `terraform import` of the plural resource is not yet implemented.
+  The FMC API supports quick bulk creation for this resource, but the deletion/modification is done one-by-one.
+  Updating and deleting `fmc_network_groups` can thus take much more time than creating it (even >500 times more time,
+  i.e. >50000%, depending on the change size).
+ds_description: >-
+  This plural data source can query a bulk of Network Groups.
+doc_category: Objects
+attributes:
+  - model_name: items
+    type: Map
+    description: >-
+      Map of network groups. The key of the map is the name of the individual Network Group. Renaming Network Groups
+      is not yet implemented.
+    map_key_example: net_group_1
+    attributes:
+      - model_name: id
+        type: String
+        resource_id: true
+        description: UUID of the managed Network Group.
+        exclude_example: true
+        exclude_test: true
+      - model_name: description
+        type: String
+        description: Optional user-created description.
+        example: "My Network Group 1"
+      - model_name: overridable
+        type: Bool
+        description: Indicates whether object values can be overridden.
+        exclude_example: true
+        test_value: "true"
+      - model_name: group_names
```

Comment:
This attribute is mainly useful for our module, where child groups are specified by name instead of a UUID; outside of this special use case users might be better off not using it. It seems `child_network_groups` would mislead users into abusing it. As the description says "Set of names (not UUIDs) of child Network Groups. [...] This is an auxiliary way to add a child Network Group: the suggested way is to add it inside `objects` by its UUID." Other attributes are also named `literals` and `objects`, not `child_literals` or `child_objects`.

Can I phrase the names/descriptions better while accenting that it's *by-name* instead of by-uuid?

Reply:
I believe the description perfectly describes how this is supposed to be used. My only concern was the attribute name, where we typically try to have consistent names when referencing other objects/resources. In this case the corresponding resource is named "Network Group" which is why I would propose `network_groups` instead of just `groups`. The convention we are following so far is to avoid a `_name(s)` suffix as this is typically the default type of reference (*by-name*). In case we are referencing another object by its ID we would add an `_id(s)` suffix to the attribute name. We can drop the `child_` prefix, so we would then have `network_groups`.

Reply:
OK, I'm too invested by now to judge. Pushed `network_groups` rename.